lib/tests/message.test.ts

```diff
@@ -347,4 +355,610 @@ describe("Send message test", () => {
+    expect(filesReceived[0].url).toBeDefined()
+    expect(filesReceived[0].type).toBe("image/jpg")
+
+    disconnect()
+  }, 30000)
+
+  test("should send 3 messages with proper delays and verify rate limiter", async () => {
+    const timeout = 1000
+    const factor = 2
+
+    const chat = await createChatInstance({
+      shouldCreateNewInstance: true,
+      config: { rateLimitFactor: factor, rateLimitPerChannel: { public: timeout } },
+    })
+
+    const channel = await chat.createPublicConversation({
+      channelId: `channel_${makeid()}`,
+      channelData: {
+        name: "Test Channel",
+        description: "This is a test channel",
+      },
+    })
+
+    const start = performance.now()
+
+    await channel.sendText("Message 1")
+
+    await channel.sendText("Message 2")
+    const durationSecond = performance.now() - start
+
+    await channel.sendText("Message 3")
+    const durationThird = performance.now() - start
+
+    expect(durationSecond).toBeGreaterThan(timeout)
+    expect(durationThird).toBeGreaterThan(timeout + timeout * factor)
+  })
+
+  test("should send long messages and validate correct rendering", async () => {
+    const messages: string[] = []
+    const longMessages = [
+      "This is a long message with a lot of text to test the rendering of long messages in the chat.",
+      "Another long message that should be rendered correctly without any issues.",
+      "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Sed non arcu eget risus lacinia tincidunt ut non orci. Nullam scelerisque odio vel erat feugiat placerat.",
+      "A very lengthy message to check how the chat handles extremely long text messages. It should not break the layout or cause any issues.",
+    ]
+
+    const disconnect = channel.connect((message) => {
+      if (message.content.text !== undefined) {
+        messages.push(message.content.text)
+      }
+    })
+
+    for (const longMessage of longMessages) {
+      await channel.sendText(longMessage)
+      const sleep = (ms: number) => new Promise((r) => setTimeout(r, ms))
+      await sleep(2000)
+    }
+
+    await waitForAllMessagesToBeDelivered(messages, longMessages)
+
+    for (const longMessage of longMessages) {
+      expect(messages).toContain(longMessage)
+    }
+
+    disconnect()
+  }, 30000)
+
+  test("should fail to send an empty or whitespace-only message", async () => {
+    const messages: string[] = []
+
+    const disconnect = channel.connect((message) => {
+      if (message.content.text !== undefined) {
+        messages.push(message.content.text)
+      }
+    })
+
+    try {
+      await channel.sendText("")
+    } catch (error) {
+      expect(error).toBeDefined()
+      expect(error.message).toContain("Message text cannot be empty")
+    }
+
+    try {
+      await channel.sendText("   ")
+    } catch (error) {
+      expect(error).toBeDefined()
+      expect(error.message).toContain("Message text cannot be empty")
```

Comment:
I don't think this test really enters this catch clock :P

Reply:
imho you should define a variable somewhere above and check the error message at the end of the test e.g.:
```
let errorMessage = ""
 try {
      await channel.sendText("   ")
    } catch (error) {
      errorMessage = error.message
    }
    
 expect(errorMessage).toContain("Message text cannot be empty")
```

Reply:
yep, thnx fixed